File: samples/nodejs-typescript-food-catalog/src/common/ContentMessage.ts
export type CrawlType = 'full' | 'incremental' | 'removeDeleted';

export type ItemAction = 'update' | 'delete';

export interface ContentMessage {
  action?: 'crawl' | 'item';
  crawlType?: CrawlType;
  itemAction?: ItemAction;
  itemId?: string;
}

File: samples/dotnet-csharp-graphdocs-ttk/Messages/ContentMessage.cs
namespace GraphDocsConnector.Messages
{
    public enum CrawlType
    {
        Full,
        Incremental,
        RemoveDeleted
    }

    internal enum ItemAction
    {
        Update,
        Delete
    }

    internal enum ContentAction
    {
        Crawl,
        Item
    }

    internal class ContentMessage
    {
        public ContentAction? Action { get; set; }
        public CrawlType? CrawlType { get; set; }
        public ItemAction? ItemAction { get; set; }
        public string? ItemId { get; set; }
    }
}
